dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactory.java

```diff
@@ -273,5 +280,68 @@ public abstract class CategoryFactory {
      * @throws DotDataException
      */
     abstract protected List<Category> getAllChildren(Categorizable parent) throws DotDataException;
+
+	/**
+	 * Return a {@link Category} Collection looking through the entire category tree starting from a specified inode.
+	 * This means the search will begin from the specified inode category and then proceed recursively through its children.
+	 *
+	 * @param searchCriteria Search Criteria
+	 *
+	 * @return List of Category filtered
+	 */
+	public abstract Collection<Category> findAll(final CategorySearchCriteria searchCriteria) throws DotDataException;
+
+	/**
+	 * Represents Search Criteria for {@link Category} searching, you cans set the follow:
+	 *
+	 * - filter: Value used to filter the Category by, returning only Categories that contain this value in their key, name, or variable name.
+	 * - inode: Entry point on the Category tree to start the searching.
+	 * - orderBy: Field name to order the Category
+	 * - direction: Order by direction, it can be 'ASC' or 'DESC'
+	 */
+	public static class CategorySearchCriteria {
+		final String inode;

```

Comment:
let's rename to `rootCategoryInode`

Reply:
done https://github.com/dotCMS/core/pull/29021/commits/15aa5848cb228fd92ea4d68fe0f1580e8a301065#diff-88379e3d0d9b5a5c54cb2207518bb686c3a9243cf6419ff11377616a7cd88dd2R303